src/OrbitQt/MultipleOfSpinBox.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef ORBIT_QT_Q_MULTIPLE_OF_EIGHT_SPIN_BOX_H_
+#define ORBIT_QT_Q_MULTIPLE_OF_EIGHT_SPIN_BOX_H_
+
+#include <QSpinBox>
+#include <memory>
+
+namespace orbit_qt {
+
+// A specialization of `QIntValidator` that only accepts values that are a multiple of
+// `multiple_of_`.
+class MultipleOfValidator : public QIntValidator {
+ public:
+  explicit MultipleOfValidator(QObject* parent = nullptr) : QIntValidator(parent) {}
+  MultipleOfValidator(int bottom, int top, int multiple_of, QObject* parent = nullptr)
+      : QIntValidator(bottom, top, parent), multiple_of_{multiple_of} {}
+  void setMultipleOf(int multiple_of) { multiple_of_ = multiple_of; }
+  [[nodiscard]] QValidator::State validate(QString& input, int& pos) const override;
+
+ private:
+  int multiple_of_ = 1;
+};
+
+// A `QSpinBox` that takes a custom validator. If the `validator_` is not set (nullptr), it will
+// fallback to `QSpinBox::validate`.
+class MultipleOfSpinBox : public QSpinBox {
+ public:
+  explicit MultipleOfSpinBox(QWidget* parent = nullptr) : QSpinBox(parent) {}
+  void SetValidator(std::unique_ptr<QIntValidator> validator) { validator_ = std::move(validator); }
```

Comment:
I'm not sure if we should have this Setter. The problem I see is that `QSpinBox` has `setRange/setMinimum/setMaximum` which doesn't compose well with a custom validator. It will be unclear to the user that these don't compose.

I would suggest to keep an instance of `MultipleOfValidator` internally in `MultipleOfSpinBox` and offer a `SetMultipleOf` setter here as well. This way it can't be confusing to the user of the class and we still get the benefit of properly abstracting the validation code into a validator class. WDYT?